tests/sentry/deletions/test_group.py

```diff
@@ -182,32 +192,100 @@ def test_delete_groups_delete_grouping_records_by_hash(
 
 
 class DeleteIssuePlatformTest(TestCase, SnubaTestCase, OccurrenceTestMixin):
-    def test_issue_platform(self):
-        event = self.store_event(data={}, project_id=self.project.id)
-        issue_occurrence, group_info = self.process_occurrence(
+    referrer = "testing.test"
+
+    def create_occurrence(
+        self, event: Event, type_id: int
+    ) -> tuple[IssueOccurrence, GroupInfo | None]:
+        occurrence, issue_platform_group = self.process_occurrence(
             event_id=event.event_id,
-            project_id=self.project.id,
-            # We are using ReplayDeadClickType as a representative of Issue Platform
-            type=ReplayDeadClickType.type_id,
-            event_data={
-                "fingerprint": ["issue-platform-group"],
-                "timestamp": before_now(minutes=1).isoformat(),
-            },
+            project_id=event.project.id,
+            type=type_id,
+            event_data={},
+        )
+        return occurrence, issue_platform_group
+
+    def select_issue_platform_events(self, columns: list[str], project_id: int) -> object:
+        return self.select_rows(Entity(EntityKey.IssuePlatform.value), columns, project_id)
+
+    def select_rows(self, entity: Entity, columns: list[str], project_id: int) -> object:
+        now = datetime.now()
+        select = [Column(column) for column in columns]
+        where = [
+            Condition(Column("project_id"), Op.IN, Function("tuple", [project_id])),
+            Condition(Column("timestamp"), Op.GTE, now - timedelta(days=1)),
+            Condition(Column("timestamp"), Op.LT, now + timedelta(days=1)),
+        ]
+        query = Query(match=entity, select=select, where=where)
+        request = Request(
+            dataset=Dataset.IssuePlatform.value,
+            app_id=self.referrer,
+            query=query,
+            tenant_ids=self.tenant_ids,
         )
+        results = bulk_snuba_queries([request])[0]["data"]
+        return results
+
+    def assert_query_n_times(
+        self, query_func: Callable[[], Any], expected_response: Sequence[object]
+    ) -> None:
+        """Query Snuba N times and assert that the response is equal to expected_response."""
+        max_retries = 10
+        retry_delay = 0.3  # seconds
+
+        for attempt in range(max_retries):
+            rows = query_func()
+            if rows == expected_response:
+                break
+            if attempt < max_retries - 1:
+                sleep(retry_delay)
+        else:
+            self.fail(
+                f"Expected rows not found in Snuba after {max_retries} attempts. Last result: {rows}"
+            )
```

Comment:
Why do we assert this n times? Are you working around the snuba query caching?

I'd recommend just adding a random condition to the query, like `0 < <random_int>`. This works since snuba caches exact queries. Or with one of your timestamps: `Condition(Column("timestamp"), Op.GTE, now - timedelta(days=1, microseconds=random.randint(0, 100000000)))`

Here's an example of where we actually have to do this in prod https://github.com/getsentry/sentry/blob/3b69278815573514cdf1319c67976d7bbd82b8a5/src/sentry/eventstore/snuba/backend.py#L386-L391

Reply:
@wedamija, it is not the caching (I will test it one more time) since all the parameters in the query are the same, thus, it should always return the same data.
It seems to simply take a moment for what was written to be ready to be served back.